### ISSUE TITLE
Pipe networks nulled/deleted from pipe parents after explosions/etc

### ISSUE THREAD

**Issue by Talkarcabbage** [Aurorastation/Aurora.3#8326]
Certain situations - so far I've just found it happening via explosion damage happening to a pipe on the network - are causing the pipe networks of atmospheric pipes (mainly the supply pipes are a concern) to be deleted AND/OR causing the pipenet to be set to null on their respective parent pipes.

This seem to then be causing a more annoying bug that results in vents, pumps, and so forth no longer pushing air to fill rooms anywhere or to refill the supply lines until the pipenet is recreated by wrenching a piece/vent up and back down (which appears to recreate the pipenet).

Probably the cause of #4744 but I felt it was worth posting more accurate and recent information, considering the original issue is two years old. 

This does not appear to have been fixed by #7766 but I assume it was either related or an attempt to do so.

I was able to produce it on a test server (multiple times) by venting a room (via an ash tile, then filling the tile to allow the room to refill) and then creating a small explosion with the bomb drop admin verb that destroyed a supply pipe. Each time a supply pipe was destroyed in this way, the vent refilling the room would drop to 0L/s and the pressure of the supply line would freeze until the vent was wrenched and put back down, at which point both the pump in atmospherics-to-distro and the vent resumed functioning normally. 

tl;dr Pipenets set to null on pipe sections (e.g. distro) after explosions/possibly other situations, seemingly breaking everything until a piece is detached/reattached

**Comment by JohnWildkins** [comment]
Definitely still an issue, is being fixed by the pipenet rework of @mikomyazaki.
Closing as duplicate of #4744 (but the extra debug information is very much appreciated)
